packages/runner-shared/src/iframe/iframe-model.js

```diff
@@ -216,7 +218,32 @@ export class IframeModel {
     this.state.messageType = 'warning'
   }
 
-  _storeOriginalState () {
+  _storeOriginalState (snapshotUrl) {
+    if (!this.isAUTSameOrigin()) {
+      const Cypress = eventManager.getCypress()
+
+      /**
+       * This only happens if the AUT ends in a cross origin state that the primary doesn't have access to.
+       * In this case, the final snapshot request from the primary is sent out to the cross-origin spec bridges.
+       * The spec bridge that matches the origin policy will take a snapshot and send it back to the primary for the runner to store in originalState.
+       */
+      Cypress.primaryOriginCommunicator.toAllSpecBridges('generate:final:snapshot', snapshotUrl)
```

Comment:
is `snapshot` different than `this.state.url`?

Reply:
I think `snapshotUrl` and `this.state.url` are the same at time of the final snapshot generation. I should be able to remove `snapshotUrl` and just pass in the state url

Reply:
updated in [c264f20](https://github.com/cypress-io/cypress/pull/20949/commits/c264f206d6ffcdfec536f4e2800f20b9a59bfcf6)